### PR TITLE
[DEPRECATION]: use factoryFor

### PR DESCRIPTION
**Pull request by snewcomer**
http://emberjs.com/deprecations/v2.x/#toc_migrating-from-_lookupfactory-to-factoryfor

Fixes::

Ref #72

### CODE REVIEW

**Comment by jasonmit** [comment]
CI failing for other reasons, this looks good - thanks!

@toranb are we targeting Node 0.12 still or can I pin Travis to >= 4?  ember-cli ecosystem dropped support for 0.12 which is the reason for the CI failure.

**Comment by toranb** [comment]
@jasonmit great point - hey @snewcomer you mind updating the travis configuration to drop legacy node quick? much appreciated :)

https://github.com/ember-redux/ember-redux/blob/master/.travis.yml#L4

above ^^ is an example of using nodeJS 4 instead of what we have here today

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
       "url": "https://github.com/toranb/ember-cli-simple-store"
   },
   "engines": {
-    "node": ">= 0.10.0"
+    "node": ">= 0.12.0"
```
Comment (jasonmit) [inline package.json]: Lets make this >= 4.0.0 also .. not that we do much in node land explicitly but to avoid any confusion

**Comment by jasonmit** [comment]
CI failing for reasons unrelated to this PR against beta and canary so I'm going to go ahead and merge.  Thanks!

**Comment by toranb** [comment]
@jasonmit huh, well normally I'd take the time to look into why the beta is failing / etc :)

@snewcomer would you be up to help nail down what broke in ember 2.12+? If not, I should free up early next month to see what broke and land a PR (mostly curious if you've seen or had issues using this addon with ember 2.12+)

**Comment by jasonmit** [comment]
@toranb master is failing with the same reason pre-this-change (it's happening against most of my addons so likely a breaking change).  I was planning to look at it separately.

**Comment by toranb** [comment]
@snewcomer @jasonmit thank you both for the effort on this! I'm thankful for the community surrounding this addon! To confirm we are up to date now and have no deprecation warnings blowing up the console => I ran the build (using ember-cli v2.10.1) and both ember v2.12 / v2.13.0-beta.1 

screen shot 2017-03-19 at 4 08 08 pm

**Pull request merged**